src/codemagic/tools/_app_store_connect/actions/publish_action.py

```diff
@@ -136,6 +140,8 @@ def _handle_ipa_testflight_submission(
             self._submit_beta_build_localization_infos(build, beta_build_infos)
         if submit_to_testflight:
             self._submit_build_to_testflight(build, app, max_processing_minutes)
+        if beta_group_names:
```

Comment:
This requires build to be submitted to TestFlight. But currently nothing checks that.